openmls/src/group/public_group/staged_commit.rs

```diff
@@ -0,0 +1,467 @@
+use super::{super::errors::*, diff::apply_proposals::ApplyProposalsValues, *};
+use crate::{
+    ciphersuite::signable::Verifiable,
+    framing::{mls_auth_content::AuthenticatedContent, mls_content::FramedContentBody, Sender},
+    group::{
+        core_group::proposals::{ProposalQueue, ProposalStore},
+        staged_commit::{MemberStagedCommitState, StagedCommitState},
+        StagedCommit,
+    },
+    messages::{proposals::ProposalOrRef, Commit},
+    schedule::{
+        psk::PskSecret, EpochSecrets, GroupEpochSecrets, InitSecret, JoinerSecret, KeySchedule,
+    },
+    treesync::node::{
+        encryption_keys::EncryptionKeyPair,
+        leaf_node::{LeafNodeTbs, TreeInfoTbs, VerifiableLeafNode},
+    },
+};
+use std::collections::HashSet;
+
+pub(crate) struct PrivateGroupParams<'a> {
```

Comment:
I'm not sure I understand this.
You are passing secrets into the public group. Isn't the idea of the public group to operate only on the public parts?